cli/src/index.ts

```diff
@@ -15,63 +23,249 @@ export type RootCommand = typeof rootCommand;
+      default: config?.jiraConsumerSecret,
     })
-    .group(["jiraConsumerKey", "jiraConsumerSecret"], authOptionGroup)
-    .demandOption(["jiraHost", "jiraConsumerSecret"]);
+    .group([jiraHostOptionKey, jiraProtocolOptionKey], "Common Required:");
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export const withAuthenticationOptions = <C extends RootCommand>(command: C) =>
+  withCommonOptions(command)
+    .group([jiraConsumerKeyOptionKey, jiraConsumerSecretOptionKey], "Auth:")
+    .demandOption([jiraHostOptionKey, jiraConsumerSecretOptionKey]);
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
 export const withAuthOptions = <C extends RootCommand>(command: C) =>
   withCommonOptions(command)
-    .option("accessToken", {
+    .option(jiraAccessTokenOptionKey, {
       alias: "t",
       type: "string",
       describe: "access token",
+      default: config?.accessToken,
     })
-    .option("accessSecret", {
+    .option(jiraAccessSecretOptionKey, {
       alias: "s",
       type: "string",
       describe: "access secret",
+      default: config?.accessSecret,
+    })
+    .option(jiraUsernameOptionKey, {
+      alias: "u",
+      type: "string",
+      describe: "your jira username",
+      default: config?.username,
+    })
+    .option(jiraPasswordOptionKey, {
+      alias: "z",
+      type: "string",
+      describe: "your jira password",
+      default: config?.password,
+    })
+    .option(jiraPersonalAccessTokenOptionKey, {
+      alias: "k",
+      type: "string",
+      describe: "your jira personal access token",
+      default: config?.personalAccessToken,
+    })
+    .group(
+      [
+        jiraConsumerKeyOptionKey,
+        jiraConsumerSecretOptionKey,
+        jiraAccessTokenOptionKey,
+        jiraAccessSecretOptionKey,
+      ],
+      "Connection - OAuth:"
+    )
+    .group(
+      [jiraUsernameOptionKey, jiraPasswordOptionKey],
+      "Connection - User Credentials:"
+    )
+    .group(
+      [jiraPersonalAccessTokenOptionKey],
+      "Connection - Personal Access Token:"
+    )
+    .option("jira", {
+      describe: "Provide at least one full set of connection parameters",
     })
-    .group(["accessToken", "accessSecret"], authOptionGroup)
-    .demandOption(["accessToken", "accessSecret"]);
+    .coerce(
+      "jira",
+      (jira: {
+        readonly protocol?: "http" | "https";
+        readonly host?: string;
+        readonly consumerKey?: string;
+        readonly consumerSecret?: string;
+        readonly accessToken?: string;
+        readonly accessSecret?: string;
+        readonly personalAccessToken?: string;
+        readonly username?: string;
+        readonly password?: string;
+      }): JiraClient => {
+        // eslint-disable-next-line functional/no-conditional-statement
+        if (jira.protocol === undefined) {
+          // eslint-disable-next-line functional/no-throw-statement
+          throw new Error(
```

Comment:
This was the best way I could find to handle multiple methods for authenticating. The only other option I could thing of was to use connection strings. e.g.

```sh
jiralint --conn oath:<consumer-name>:<consumer-secret>:<access-token>:<access-secret> search -j <jql>
jiralint --conn pat:<personal-access-token> search -j <jql>
jiralint --conn credentials:<username>:<password> search -j <jql>
```